src/repositories/user.repository.ts

```diff
@@ -65,15 +88,20 @@ export async function remove(userId: string): Promise<{ affectedRows: number }>
  */
 export async function findAll(options: Filters = {}): Promise<{ data: User[], totalCount: number }> {
   const allOptions = Object.assign({}, defaultFilters, options);
-  const searchFields = ['id', 'email', 'firstName', 'lastName'];
-  const sortFields = ['email', 'firstName', 'lastName', 'role', 'hasAccess', 'registrationCompleted'];
+  const searchFields = [`${aliases.USERS}.id`, 'email', 'firstName', 'lastName'];
+  const sortFields = ['email', 'firstName', 'lastName', 'role', `${aliases.STATUSES}.code`];
 
-  const query = selectAndCount(db, defaultReturnValues)
-    .from(tableNames.USERS);
+  const query = selectAndCount(db, extendedReturnValues)
+    .from(`${tableNames.USERS} as ${aliases.USERS}`)
+    .join(`${viewNames.USER_STATUSES} as ${aliases.STATUSES}`, `${aliases.USERS}.status`, `${aliases.STATUSES}.id`);
 
   applyPagination(query, allOptions);
   applySearch(query, allOptions, searchFields);
-  applySorting(query, allOptions, sortFields);
+
+  // if sortField is status, replace status with us.code for sorting on status code ACTIVE OR INACTIVE
```

Comment:
Mistake in comment **ACTIVE OR INACTIVE** = **COMPLETE_REGISTRATION, REGISTERED, BLOCKED**